app/containers/ItunesContainer/index.js

```diff
@@ -0,0 +1,200 @@
+
+const { Search } = Input;
+
+const CustomCard = styled(Card)`
+  && {
+    margin: 20px 0;
+    max-width: ${props => props.maxwidth};
+    color: ${props => props.color};
+    ${props => props.color && `color: ${props.color}`};
+  }
+`;
+const Container = styled.div`
+  && {
+    display: flex;
+    flex-direction: column;
+    max-width: ${props => props.maxwidth}px;
+    width: 100%;
+    margin: 0 auto;
+    padding: ${props => props.padding}px;
+  }
+`;
+const RightContent = styled.div`
+  display: flex;
+  align-self: flex-end;
+`;
+export function ItunesContainer({
+  dispatchItunesSongs,
+  dispatchClearItunesSongs,
+  intl,
+  songsData = {},
+  songsError = null,
+  songName,
+  maxwidth,
+  padding
+}) {
+  useInjectSaga({ key: 'itunesContainer', saga });
+
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    const loaded = get(songsData, 'results', null) || songsError;
+    if (loading && loaded) {
+      setLoading(false);
+    }
+  }, [songsData]);
+
+  useEffect(() => {
+    if (songName && !songsData?.results?.length) {
+      dispatchItunesSongs(songName);
+      setLoading(true);
+    }
+  }, []);
+
+  const history = useHistory();
+
+  const handleOnChange = rName => {
+    if (!isEmpty(rName)) {
+      dispatchItunesSongs(rName);
+      setLoading(true);
+    } else {
+      dispatchClearItunesSongs();
+    }
+  };
+  const debouncedHandleOnChange = debounce(handleOnChange, 200);
+
+  const renderRepoList = () => {
+    const items = get(songsData, 'results', []);
+    const resultCount = get(songsData, 'resultCount', 0);
+    return (
+      (items.length !== 0 || loading) && (
+        <CustomCard>
+          <Skeleton loading={loading} active>
+            {songName && (
+              <div>
+                <T id="search_query" values={{ songName }} />
+              </div>
+            )}
+            {resultCount !== 0 && (
+              <div>
+                <T id="matching_results" values={{ resultCount }} />
+              </div>
+            )}
+            {items.map((item, index) => (
+              <CustomCard key={index}>
+                <T id="song_name" values={{ name: item.trackName }} />
+                <T id="song_artist" values={{ fullName: item.artistName }} />
+                <T id="release_date" values={{ stars: item.releaseDate }} />
+              </CustomCard>
+            ))}
+          </Skeleton>
+        </CustomCard>
+      )
+    );
+  };
+  const renderErrorState = () => {
+    let repoError;
+    if (songsError) {
+      repoError = songsError;
+    } else if (!get(songsData, 'resultCount', 0)) {
+      repoError = 'song_search_default';
+    }
+    return (
+      !loading &&
+      repoError && (
+        <CustomCard color={songsError ? 'red' : 'grey'} title={intl.formatMessage({ id: 'song_list' })}>
+          <T id={repoError} />
+        </CustomCard>
+      )
+    );
+  };
+  const refreshPage = () => {
+    history.push('itunes');
```

Comment:
Let's use route constants here

Reply:
Also please refer the document I shared for branch naming conventions and PR prerequisite.